feat(india): E Invoice Cancel Eway Bill is enabled.

maharshivpatel opened the pull request:
Adaequare has enabled E-Way Bill Cancellation on thier API so necessary changes are made to enable functionality for the same.

This closes #30887 

`no-docs`

maharshivpatel committed:
feat(india): E Invoice Cancel Eway Bill enabled.

Adaequare has enabled E-Way Bill Cancellation on thier API so neccesory changes are made to enable functionality for the same.

maharshivpatel committed:
removed comment to avoid linter errors

i have removed comment to fix linter error.

nextchamp-saqib commented on erpnext/regional/india/e_invoice/utils.py:
```diff
@@ -1300,10 +1299,9 @@ def generate_eway_bill(doctype, docname, **kwargs):
+	gsp_connector = GSPConnector(doctype, docname)
+	gsp_connector.cancel_eway_bill(eway_bill, reason, remark)
 
 	frappe.db.set_value(doctype, docname, "ewaybill", "")
```
These 2 set_value are unnecessary now, these will be updated after the API request is successful

maharshivpatel replied:
done

nextchamp-saqib commented on erpnext/regional/india/e_invoice/utils.py:
```diff
@@ -1300,10 +1299,9 @@ def generate_eway_bill(doctype, docname, **kwargs):
+	gsp_connector = GSPConnector(doctype, docname)
+	gsp_connector.cancel_eway_bill(eway_bill, reason, remark)
 
 	frappe.db.set_value(doctype, docname, "ewaybill", "")
```
These 2 set_value are unnecessary now, these will be updated after the API request is successful

maharshivpatel committed:
unnecessary set_value removed

I have removed set_value as they are unnecessary and will be updated after a successful API request.

maharshivpatel committed:
Merge branch 'frappe:develop' into feat-india-e_invoice-cancel-eway-bill

maharshivpatel committed:
Merge branch 'frappe:develop' into feat-india-e_invoice-cancel-eway-bill

nextchamp-saqib committed:
fix: linter

nextchamp-saqib commented on erpnext/regional/india/e_invoice/einvoice.js:
```diff
@@ -136,27 +136,58 @@ erpnext.setup_einvoice_actions = (doctype) => {
+								},
+								error: () => {
+									frappe.show_alert({
+										message: __('E-Way Bill was not Cancelled '),
```
```suggestion
										message: __('E-Way Bill was not Cancelled'),
```

nextchamp-saqib committed:
fix: linter
